tests/api/runtimes/test_spark.py

```diff
@@ -465,16 +484,18 @@ def test_with_node_selector(self, db: sqlalchemy.orm.Session, k8s_secrets_mock):
         self._assert_merged_node_selectors(
             body,
             mlrun.utils.helpers.merge_with_precedence(
-                self.project_default_function_node_selector, node_selector
+                project_node_selector, node_selector
             ),
             mlrun.utils.helpers.merge_with_precedence(
-                self.project_default_function_node_selector, driver_node_selector
+                project_node_selector, driver_node_selector
             ),
             mlrun.utils.helpers.merge_with_precedence(
-                self.project_default_function_node_selector, executor_node_selector
+                project_node_selector, executor_node_selector
             ),
         )
 
+        self.project = default_project_name
```

Comment:
this is an issue because if the test fails than this will not be set back. After we found the original issue I believe you should be able to use the same project that is created in the setup